server/client.go

```diff
@@ -3586,15 +3586,19 @@ func removeHeaderIfPresent(hdr []byte, key string) []byte {
 	if start < 0 {
 		return hdr
 	}
+	index := start + len(key)
+	if index >= len(hdr) || hdr[index] != ':' {
+		return hdr
+	}
```

Comment:
Is it possible to add hdr[start -1] == '\n' ? 
Keys start at the beginning of a line.

Reply:
ok take one more look, thanks..

Reply:
Good idea, done.